src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,881 @@
+    ad.user("user").add()
+    user1 = ad.user("user1").add()
+    user2 = ad.user("user2").add()
+    deny_user1 = ad.user("deny_user1").add()
+    deny_user2 = ad.user("deny_user2").add()
+    group = ad.group("group").add().add_members([user2])
+    deny_group = ad.group("deny_group").add().add_members([deny_user2])
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, group, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [deny_user1, deny_group],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.start()
+
+    assert not client.auth.parametrize(method).password(username="user", password="Secret123")
+    assert client.auth.parametrize(method).password(username="user1", password="Secret123")
+    assert client.auth.parametrize(method).password(username="user2", password="Secret123")
+    assert not client.auth.parametrize(method).password(username="deny_user1", password="Secret123")
+    assert not client.auth.parametrize(method).password(username="deny_user2", password="Secret123")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["su", "ssh"])
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_enforcing_with_no_policy(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host base access control is set to enforcing with no policy
+    :description:
+        When ad_gpo_access_control is set to enforcing with no policy applied. It should
+        permit all users to login.
+    :setup:
+        1. Create the following user, 'user'
+        2. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing'
+        3. Start SSSD
+    :steps:
+        1. Authenticate 'user'
+        2. Check logs for gpo_access_check result
+    :expectedresults:
+        1. User authentication is successful
+        2. Access check result is granted
+    :customerscenario: True
+    """
+    ad.user("user").add()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.start()
+
+    assert client.auth.parametrize(method).password(username="user", password="Secret123")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["su", "ssh"])
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_permissive_and_users_are_allowed(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host base access control is set to permissive
+    :description:
+        When ad_gpo_access_control is set to permissive, all users are able to login despite what
+        is configured in the policy. The rules are still processed and can be viewed in the logs.
+    :setup:
+        1. Create the following user, 'user1'
+        2. Create and link the GPO 'site policy' and add 'user1', group and 'Domain Admins' to
+           SeInteractiveLogonRight key. Add 'deny_user1' to SeDenyInteractiveLogonRight key
```

Comment:
Hi,

I think `Add 'deny_user1' to SeDenyInteractiveLogonRight key` can be removed here.

bye,
Sumit